### DISCUSSION TITLE
Controlling window in desktop

### DISCUSSION THREAD

**Discussion by l-e-webb** [nacular/doodle#42] [Ideas]
I've been really digging Doodle, and currently I'm using it to prototype a game to run on desktop and in browser. One thing I need is the ability to control window size on desktop. I'm currently using a hack where I added this to the application creator block in my jvm module:
```
instance<JFrame>().apply {
    isResizeable = false
    preferredSize = Dimension(...)
    // etc
    pack()
}
```

I'm wondering: is there a better way to do this? And, if not, would there be a good, idiomatic way to make this functionality available in Doodle via some abstraction? I'm not sure where it'd belong, given Doodle's focus on minimizing platform-specific code. If anyone can give me pointers on how to fit it in appropriately, I'd be happy to try to put a PR together.

**Comment by pusolito** [comment]
Glad to hear about your project! I haven't decided on an approach for this yet, so your work-around is reasonable. I suspect the right way to do this will be via the application block though. This block is platform-specific and can allow for this sort of thing.

Let me know if you have any thoughts and I'll keep this in mind for the next round of desktop updates.

I'm also interested to hear your feedback in general. What's working well and what could be better?

Thanks.

**Comment by pusolito** [comment]
i just shared an [update](https://kotlinlang.slack.com/archives/C01CJM07MGV/p1705076786701079) in Doodle's Kotlin Slack channel ([access here](http://slack.kotl.in/)) about progress toward 0.10.0.

there is a video showing an early prototype for multi-window support. apps would have access to a new type for window management that enables customization and creation of windows, creating/managing OS menu bars etc.. this work is still early, but wanted to share the direction so far.

**Reply by pusolito** [reply]
You can now do this in the latest release: 0.10.0.
